helix-view/src/document.rs

```diff
@@ -686,6 +697,13 @@ impl Document {
         config_loader: Option<Arc<ArcSwap<syntax::Loader>>>,
         config: Arc<dyn DynAccess<Config>>,
     ) -> Result<Self, Error> {
+        // if the path is not a regular file (e.g.: /dev/random) it should not be opened
+        if path.exists() && !path.is_file() && !path.is_symlink() {
```

Comment:
this is quite inefficient since it requires 3 syscalls, you can do it in one call with `path.metadata().map_or(false, |metadata| !metadata.is_file())`

Reply:
Good catch, the suggestions are implemented